### PR TITLE
RestartDriver() API + Common code for PerfromTask

### PR DESCRIPTION
**Pull request by vinayakshnd**
1. Added RestartDriver() API to volume driver.
2. Factored out common code for trigger-based perform task action in
a separate package.

PX-Backup Jenkins job which is consuming changes from this PR:
https://jenkins.portworx.co/job/tp-backup-vinayak/18/console
Test Suite Name: BackupCreateRestore

### CODE REVIEW

File: drivers/volume/portworx/schedops/k8s-schedops.go [hunk 1]
```diff
@@ -2,11 +2,12 @@ package schedops
 
 import (
 	"fmt"
-	"k8s.io/apimachinery/pkg/version"
 	"regexp"
 	"strings"
 	"time"
 
+	"k8s.io/apimachinery/pkg/version"
+
```
Comment (harsh-px) [inline drivers/volume/portworx/schedops/k8s-schedops.go:10]: remove empty lines

File: drivers/api/api.go [hunk 2]
```diff
@@ -0,0 +1,108 @@
...
+				return false, false, err
+			}
+
+			if triggered {
+				// run user task and return
+				logrus.Debugf("[debug] %s trigger check passed. Now running user task.\n", fn)
+				return triggered, false, userTask()
+			}
+
+			return false, true, fmt.Errorf("%s: trigger check hasn'userTaskWithTriggerChecks been met yet", fn)
```
Comment (harsh-px) [inline drivers/api/api.go:67]: there is a typo in this error message string.

File: drivers/api/api.go [hunk 3]
```diff
@@ -0,0 +1,108 @@
...
+		userTaskWithTriggerChecks := func() (interface{}, bool, error) {
+			triggered, err := opts.TriggerCb()
+			if err != nil {
+				logrus.Warnf("%s: failed to invoke trigger callback function due to: %v", fn, err)
+				return false, false, err
+			}
+
+			if triggered {
+				// run user task and return
+				logrus.Debugf("[debug] %s trigger check passed. Now running user task.\n", fn)
```
Comment (harsh-px) [inline drivers/api/api.go:63]: remove this debug before commit. Use for testing.

File: drivers/api/api.go (hunk 2 above)
Comment (vinayakshnd) [reply drivers/api/api.go]: Done

File: drivers/api/api.go (hunk 3 above)
Comment (vinayakshnd) [reply drivers/api/api.go]: Done

File: drivers/api/api.go [hunk 4]
```diff
@@ -0,0 +1,107 @@
+package api
+
+import (
+	"fmt"
+	"time"
+
+	"github.com/libopenstorage/openstorage/pkg/sched"
+
```
Comment (harsh-px) [inline drivers/api/api.go:8]: Extra line

**Comment by vinayakshnd** [comment]
Jenkins tp-dev-test job: https://jenkins.portworx.co/job/Torpedo/job/tp-dev-test/443/console

File: drivers/api/api.go (hunk 4 above)
Comment (vinayakshnd) [reply drivers/api/api.go]: Done

**Pull request merged**